crates/cdk/src/wallet/mod.rs

```diff
@@ -500,17 +497,34 @@ impl Wallet {
     ///     let wallet = Wallet::new(mint_url, unit, Arc::new(localstore), &seed, None)?;
     ///     let amount = Amount::from(100);
     ///
-    ///     let quote = wallet.mint_quote(amount).await?;
+    ///     let quote = wallet.mint_quote(amount, None).await?;
     ///     Ok(())
     /// }
     /// ```
     #[instrument(skip(self))]
-    pub async fn mint_quote(&self, amount: Amount) -> Result<MintQuote, Error> {
+    pub async fn mint_quote(&self, amount: Amount, description: Option<String>) -> Result<MintQuote, Error> {
         let mint_url = self.mint_url.clone();
         let unit = self.unit;
+
+        // If we have a description, we check that the mint supports it.
+        if description.is_some() {
+            let mint_method_settings = self.localstore
+                .get_mint(mint_url.clone())
+                .await?
+                .expect("Could not retrieve mint information")
+                .nuts
+                .nut04
+                .get_settings(&unit, &PaymentMethod::Bolt11)
+                .expect("Could not find mint unit and method");
```

Comment:
Can we remove the expect here and return an error `Error::UnitUnsupported` expect will panic like unwrap does so it should be avoided, unless it checked in someway so we know it won't panic.

Reply:
I tried using the `match` construct like you showed in the code style guide, but it wouldn't let me. So i used `if let`.